core/Piwik.php

```diff
@@ -282,6 +282,30 @@ public static function hasUserSuperUserAccessOrIsTheUser($theUser)
         }
     }
 
+    /**
+     * Returns if the given user needs to confirm his password in UI and for certain API methods
+     *
+     * @param string $login
+     * @return bool
+     */
+    public static function doesUserRequirePasswordConfirmation(string $login)
+    {
+        $requiresPasswordConfirmation = true;
+
+        /**
+         * Triggered to check if a password confirmation for a user is required.
```

Comment:
should we mention that this makes it less secure and poses a security risk? 

Or can we better provide a way for the plugin to check the password if that's not the case? Like Ldap plugin would check the password against LDAP. Gitlab login plugin would verify the password in GitLab. Not sure if a plugin can already do this?

It seems like `isPasswordCorrect` is already using the Auth adapter and this should work already so maybe there's a different problem?

Reply:
Can add some note regarding security. 

`isPasswordCorrect` should use the auth adapter, which should allow plugins to validate against certain services.
But there might be plugins, where you can't validate a password. Not sure how the SAML plugin handles that, but if you actually log in on an external sso and the sso only passes through the current user, there might be no way to validate a password. Guess the [LoginOIDC plugin](https://plugins.matomo.org/LoginOIDC) does something similar.

Reply:
For simplicity lets assume you use a service to log in users that doesn't use a password at all. Having a password confirmation won't work in that case, as you can't validate it. For sure you could simply accept any input, but not sure if that behavior is better.

Reply:
Got it now 👍  I guess the only issue is that this will make their install less secure.

Would they maybe need to implement the method `confirmPassword`? https://github.com/matomo-org/matomo/blob/4.10.0/plugins/Login/PasswordVerifier.php#L192 redirects to this. Then the controller does set when the password was verified correctly. This would then maybe happen on some third party provider etc. Not sure if that would work in this case?

Reply:
@tsteur Yes. That one is the password verify that is triggered on PHP side. But guess simply doing a `$this->passwordVerify->setPasswordVerifiedCorrectly();` if it shouldn't be checked might do the trick. The changes in this PR are mostly for the password confirmation we show in the UI to send the password with API requests.

But guess it makes sense to also skip the PHP check using the event. Will add that.

Reply:
On the other side, I think it would be better to force other login plugins to overwrite `confirmPassword`. They can then decide theirself if they simply want to set the password verified, or maybe check the login state on the used sso/service again.

Reply:
👍